File: src/pub_sub/src/sub.cpp
#include"ros/ros.h"
#include"std_msgs/String.h"
#include<sstream>

using namespace std;

void callback(const std_msgs::String::ConstPtr &msg)
{
    ROS_INFO("Receive %s", msg->data.c_str());
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "sub_c");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("hello", 10, callback);
    ros::spin();
    return 0;
}

File: src/pub_sub/scripts/sub.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import String


def callback(msg):
    rospy.loginfo("Receive %s", msg.data)


if __name__ == "__main__":
    rospy.init_node("sub_py")
    sub = rospy.Subscriber("world", String, callback, queue_size=10)
    rospy.spin()